src/main.rs

```diff
@@ -40,41 +40,53 @@ fn main() -> BoxResult<()> {
 async fn async_main() -> BoxResult<()> {
     dotenv::dotenv().ok();
     println!("Booting ingestion server...");
-    let mongo_url = env::var("MONGO_URL")?;
+    let mongo_url = match env::var("MONGO_URL") {
+        Ok(url) => url,
+        _ => panic!("MONGO_URL variable not specified. Please specify the MongoDB's URL in this format: mongodb://user:password@some_mongo"),
+    };
 
     let mut client_options = ClientOptions::parse(mongo_url).await?;
     client_options.app_name = Some("Sentinel Surveillance".to_string());
-    client_options.tls = Some(Tls::Enabled(Default::default()));
+    match env::var("TLS") {
+        Ok(tls) => match tls.as_str() {
+            "true" => client_options.tls = Some(Tls::Enabled(Default::default())),
+            _ => println!("TLS Disabled"),
+        },
+        _ => println!("TLS Disabled"),
+    }
 
     let client = Client::with_options(client_options)?;
 
-    let database = client.database("imagine2022");
-
+    let database = match env::var("DATABASE") {
+        Ok(db_name) => client.database(&db_name),
+        _ => panic!("DATABASE variable not specified. Please specify DATABASE name."),
+    };
     database.run_command(doc!("ping": 1), None).await?;
 
     println!("Connected to mongo database.");
 
     let bluetooth_frames = database.collection::<BluetoothFrame>("bluetooth_frames");
 
-    let socket = UdpSocket::bind("0.0.0.0:8080").await?;
+    let socket_address = match env::var("SERVER_ADDRESS") {
+        Ok(addr) => addr,
+        _ => panic!("SERVER_ADDRESS variable not specified. Please specify the server's address and port. (eg 0.0.0.0:8080)"),
+    };
+
+    let socket = UdpSocket::bind(&socket_address).await?;
     let mut buffer = [0; 1024];
 
+    println!("Serving on {}", &socket_address);
+
     loop {
         let (len, _address) = socket.recv_from(&mut buffer).await?;
-
-        let bluetooth_frame =
-            serde_json::from_slice::<BluetoothFrameWithoutTimestamp>(&buffer[0..len]);
-
-        match bluetooth_frame {
-            Ok(bluetooth_frame) => {
-                let now = SystemTime::now();
-                let timestamp = now.duration_since(UNIX_EPOCH).unwrap();
-
+        let frame = match std::str::from_utf8(&buffer) {
+            Ok(frame) => {
+                let frame_vec: Vec<&str> = frame.split("|").collect();
                 let bluetooth_frame = BluetoothFrame {
-                    macaddr: bluetooth_frame.macaddr,
-                    uename: bluetooth_frame.uename,
-                    rssi: bluetooth_frame.rssi,
-                    timestamp: timestamp.as_millis() as u64,
+                    macaddr: frame_vec[3].to_string(),
+                    uename: frame_vec[2].to_string(),
+                    rssi: frame_vec[4].parse::<i32>().unwrap(), // This might panic. I've seen this come out of order.
+                    timestamp: frame_vec[1].parse::<u64>().unwrap()
```

Comment:
Might want to use a match statement instead of unwapping. Could also do some verification of the macaddr and uename fields too. We don't _really_ need the uename anymore as well.

Reply:
Interesting. Since we're using this simpler format new, I think we should remove `BluetoothFrameWithoutTimestamp` because its whole purpose was to be the JSON format of the data sent.

Reply:
If you're willing to try some unsafe Rust, I recommend using a try block here, or otherwise a function that returns an `Option<BluetoothFrame>`. Within that function use can use `?` instead of unwrap, and then here you can use `if let Some(frame) = parsed_frame {...}` to store the result if every option and result was good.

Reply:
Went with the second option here. What are your thoughts? I'm guessing I can get proper errors out of this somehow? For instance, I'd like to know if the `from_utf8` fails, or if the `parse::<i32>()` fails? In particular, the `.ok()?` seems off to me.